Matchismo/Matchismo/Model/CardMatchingGame.mm

```diff
@@ -114,25 +131,43 @@ - (void)chooseCardAtIndex: (NSUInteger)index {
           for (Card * c in self.chosenCards) {
             c.matched = YES;
           }
-          NSString *description = @"Matched";
+          NSMutableAttributedString *descriptionString = [[[NSAttributedString alloc]
+                                                             initWithString: @"Matched"]
+                                                          mutableCopy];
           for (Card *c in matchResult.matches) {
-            description = [description stringByAppendingString:[NSString stringWithFormat:@" %@", c.contents]];
+            [descriptionString appendAttributedString:[[NSAttributedString alloc]
+                                                         initWithString:@" "]];
+            [descriptionString appendAttributedString:c.contents];
           }
-          description = [description stringByAppendingString:[NSString stringWithFormat:@" for %d points.", points]];
-          self.lastCardFlipDescription = description;
+          [descriptionString appendAttributedString:
+             [[NSAttributedString alloc] initWithString:
+                [NSString stringWithFormat:@" for %d points.", points]]];
+          [self.chosenCards addObject:card];
+          self.lastMoveDescription = descriptionString;
           [self.chosenCards removeAllObjects];
         } else { // No match
           self.score -= MISMATCH_PENALTY;
           Card *cardToUnchoose = self.chosenCards[self.chosenCards.count-1];
           cardToUnchoose.chosen = NO;
-          [self.chosenCards removeObject:cardToUnchoose];
           [self.chosenCards insertObject:card atIndex:0];
-          self.lastCardFlipDescription = [NSString stringWithFormat:@"No match! %d point penalty!",
-                                          MISMATCH_PENALTY];
+          NSMutableAttributedString *descriptionString = [[[NSAttributedString alloc]
```

Comment:
This whole method is way too big so it's hard to see what's going on.
If you decompose it into a bunch of smaller private methods it'll be much more readable.